types/base_tx.go

```diff
@@ -105,3 +107,79 @@ func (tx *BaseTx) copy() *BaseTx {
 
 	return cpy
 }
+
+func (tx *BaseTx) unmarshalJSON(v *fastjson.Value) error {
+	hash, err := unmarshalJSONHash(v, "hash")
```

Comment:
Just to check, there isn't some nicer approach where we could unmarshal all the data at once, but we need this custom unmarshaling logic (field by field)?

Reply:
This is using the `fastjson` library for the best possible performance when unmarshalling results. Didn't want to use the built in json library because for larger strings (responses) it is just not good. We witnessed that with the Palm guys if you remembered. They had a similar fix.